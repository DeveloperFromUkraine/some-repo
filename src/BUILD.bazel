#load("@angular//:index.bzl", "ng_module")
#load("@build_bazel_rules_nodejs//:defs.bzl", "history_server", "rollup_bundle")
#load("@build_bazel_rules_nodejs//internal/web_package:web_package.bzl", "web_package")
#load("@build_bazel_rules_typescript//:defs.bzl", "ts_library", "ts_config", "ts_devserver")
#
#package(default_visibility = ["//visibility:public"])
#
## Needed because the prodserver only loads static files that appear under this
## package.
#genrule(
#    name = "zonejs",
#    srcs = ["@npm//node_modules/zone.js:dist/zone.min.js"],
#    outs = ["zone.min.js"],
#    cmd = "cp $< $@",
#)
#
#genrule(
#    name = "systemjs",
#    srcs = ["@npm//node_modules/systemjs:dist/system.js"],
#    outs = ["system.js"],
#    cmd = "cp $< $@",
#)
#
#ts_config(
#    name = "tsconfig-test",
#    src = "tsconfig.spec.json",
#    deps = [":tsconfig.json"],
#)
#
#ng_module(
#    name = "src",
#    srcs = glob(
#        include = ["**/*.ts"],
#        exclude = [
#            "**/*.spec.ts",
#            "main.ts",
#            "test.ts",
#            "initialize_testbed.ts",
#        ],
#    ),
#    assets = [
#        "//src/styles:styles",
#        "//src:index.html",
#    ],
#    tsconfig = "//:tsconfig.json",
#    deps = [
#        "@angular//packages/core",
#        "@angular//packages/platform-browser",
#        "@angular//packages/platform-browser/animations",
#        "@angular//packages/router",
#    ],
#)
#
#ts_devserver(
#    name = "devserver1",
#    port = 4200,
#    additional_root_paths = [
#        "npm/node_modules/zone.js/dist",
#    ],
#    entry_module = "ignite_design_system_2/src/main.dev",
#    serving_path = "/bundle.min.js",
#    static_files = [
#        "@npm//node_modules/zone.js:dist/zone.min.js",
#        "index.html",
#    ],
#    deps = ["//src"],
#)
#
#rollup_bundle(
#    name = "bundle",
#    entry_point = "src/main.prod",
#    deps = ["//src"],
#)
#
#ts_library(
#    name = "initialize_testbed",
#    testonly = 1,
#    srcs = [
#        "initialize_testbed.ts",
#    ],
#    deps = [
#        "@angular//packages/core/testing",
#        "@angular//packages/platform-browser-dynamic/testing",
#        "@npm//@types",
#    ],
#)
#
#ts_web_test_suite(
#    name = "test",
#    srcs = ["@npm//node_modules/tslib:tslib.js"],
#    runtime_deps = [
#        ":initialize_testbed",
#    ],
#    # do not sort
#    bootstrap = [
#        "@npm//node_modules/zone.js:dist/zone-testing-bundle.js",
#        "@npm//node_modules/reflect-metadata:Reflect.js",
#    ],
#    browsers = [
#        "@io_bazel_rules_webtesting//browsers:chromium-local",
#    ],
#    deps = [
#        ":test_lib",
#        "@npm//karma-jasmine",
#    ],
#)
#
#history_server(
#    name = "prodserver",
#    data = [
#        "index.html",
#        ":bundle",
#        ":zonejs",
#    ],
#)
