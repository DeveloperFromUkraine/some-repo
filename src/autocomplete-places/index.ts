export * from './autocomplete-places.component';
export * from './autocomplete-places.service';
export * from './autocomplete-service.provider';
