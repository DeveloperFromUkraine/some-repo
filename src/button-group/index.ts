import { Component } from '@angular/core';

@Component({
  selector: 'ign-button-group',
  template: '<ng-content></ng-content>',
  styleUrls: ['./button-group.css'],
})
export class ButtonGroupComponent {}
