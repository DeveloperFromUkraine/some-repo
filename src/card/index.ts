export * from './card-actions/index';
export * from './card-aside/index';
export * from './card-container/index';
export * from './card-header-container/index';
export * from './card-main/index';
export * from './card-section/index';
export * from './card-section-subtitle/index';
export * from './card-section-title/index';
export * from './card-subtitle/index';
export * from './card-title/index';
export * from './card-toolbar/index';
