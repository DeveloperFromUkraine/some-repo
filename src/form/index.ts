export * from './form-group';
export * from './form-sizing';
export * from './nestable-form';
