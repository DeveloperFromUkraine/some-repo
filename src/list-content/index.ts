export * from './list-content.component';
export * from './list-content-card/index';
export * from './list-content-empty-state/index';
export * from './list-content-panel/index';
export * from './list-content-unselected-state/index';
