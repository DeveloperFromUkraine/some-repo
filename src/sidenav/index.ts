export * from './sidenav-header/index';
export * from './sidenav-list/index';
export * from './sidenav-list-item/index';
