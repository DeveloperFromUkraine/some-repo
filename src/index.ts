import { NgModule, ModuleWithProviders } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { CommonModule, CurrencyPipe } from '@angular/common';
import { RouterModule } from '@angular/router';

import { PageComponent } from './page/page.component';
export * from './page/page.component';

import { RightDrawerComponent } from './right-drawer/right-drawer.component';
export * from './right-drawer/right-drawer.component';

import {
  SideNavHeaderComponent,
  SideNavListComponent,
  SideNavListItemComponent,
} from './sidenav/index';
export * from './sidenav/index';

import {
  ListItemComponent,
  ListItemContentComponent,
  ListContainerComponent,
} from './list-item/list-item.component';
export * from './list-item/list-item.component';

import {
  ListContentCardComponent,
  ListContentComponent,
  ListContentEmptyStateComponent,
  ListContentPanelComponent,
  ListContentUnselectedStateComponent,
} from './list-content/index';
export * from './list-content/index';

import {
  EmptyStateComponent,
  EmptyStateTitleComponent,
  EmptyStateDescriptionComponent,
  EmptyStateImageComponent,
} from './empty-state/empty-state.component';
export * from './empty-state/empty-state.component';

import {
  ListFilterPipe,
  ListSelectionAddButtonComponent,
  ListSelectionComponent,
  ListSelectionItemComponent,
  ListSelectionSearchComponent,
} from './list-selection/index';
export * from './list-selection/index';

import {
  CardHeaderContainerComponent,
  CardSubtitleComponent,
  CardMainComponent,
  CardAsideComponent,
  CardTitleComponent,
  CardSectionComponent,
  CardSectionSubTitleComponent,
  CardSectionTitleComponent,
  CardActionsComponent,
  CardToolbarComponent,
} from './card/index';
export * from './card/index';

import {
  ListItemPrimaryTextComponent,
  ListItemSecondaryTextComponent,
} from './text/text.component';
export * from './text/text.component';

import { SelectContainerComponent } from './select-container/select-container.component';
export * from './select-container/select-container.component';

import {
  FormGroupComponent,
  SizingFullDirective,
  SizingLargeDirective,
  SizingMediumDirective,
  SizingSmallDirective,
  NestableFormDirective,
} from './form/index';
export * from './form/index';

import { ErrorBannerComponent } from './error-banner/error-banner.component';
export * from './error-banner/error-banner.component';

import { InfoBannerComponent } from './info-banner/info-banner.component';
export * from './info-banner/info-banner.component';

import { ButtonGroupComponent } from './button-group/index';
export * from './button-group/index';

import { ExpandableFabComponent } from './expandable-fab/expandable-fab.component';
export * from './expandable-fab/expandable-fab.component';

import { ExpandableFabItemComponent } from './expandable-fab/expandable-fab-item.component';
export * from './expandable-fab/expandable-fab-item.component';
import { ExpandableFabItemModule } from './expandable-fab/expandable-fab-item.module';

import {
  FieldComponent,
  FieldInlineComponent,
  EditableFieldComponent,
  EditableFieldInlineComponent,
} from './field/field.component';
export * from './field/field.component';

import { CenterComponent } from './center/center.component';

import { LoadingContainerComponent } from './loading-container/loading-container.component';
import { FooterComponent } from './footer/footer.component';

import { DateRangeModule } from './date-range/date-range.module';
import { DateRangeComponent } from './date-range/date-range.component';

import { TextListItemModule } from './text-list-item/text-list-item.module';
import { TextListItemComponent } from './text-list-item/text-list-item.component';

import { DataTableContainerModule } from './data-table-container/data-table-container.module';
import { DataTableContainerComponent } from './data-table-container/data-table-container.component';

import { ExpandableSearchComponent } from './expandable-search/expandable-search.component';
import { RightAlignContainerComponent } from './right-align-container/right-align-container.component';

import { ListSearchComponent } from './list-search/list-search.component';

import {
  AccessibleClickDirective,
  AccessibleViewDirective,
  AccessibleRemoveViewDirective,
  AccessibleContextMenuViewDirective,
  ListItemRoleDirective,
  ToolTipDirective,
} from './accessibility/accessibility.directive';

import { TranslationService } from './localization/translation.service';

import { TranslationModule } from './localization/translation.module';
import { AvatarComponent } from './avatar/avatar.component';
import { AvatarGroupComponent } from './avatar-group/avatar-group.component';
export * from './avatar-group/avatar-group.types';

import {
  InputMaskComponent,
  DisplayModeDirective,
} from './currency-input-mask/base-input-mask/input-mask.component';
import { CurrencyMaskComponent } from './currency-input-mask/currency-mask/currency-mask.component';
import { CurrencyMaskDirective } from './currency-input-mask/currency-mask/currency-mask.directive';

import { MessagesComponent } from './messages/messages.component';
export * from './messages/messages.component';

import {
  MatToolbarModule,
  MatMenuModule,
  MatButtonModule,
  MatCardModule,
  MatIconModule,
  MatRippleModule,
  MatProgressSpinnerModule,
  MatNativeDateModule,
  MatListModule,
  MatFormFieldModule,
  MatInputModule,
  MatTooltipModule,
  MatSidenavModule,
} from '@angular/material';

const components = [
  CardHeaderContainerComponent,
  CardSubtitleComponent,
  PageComponent,
  RightDrawerComponent,
  SideNavHeaderComponent,
  SideNavListComponent,
  SideNavListItemComponent,
  ListContainerComponent,
  ListContentComponent,
  ListContentCardComponent,
  ListItemComponent,
  ListItemContentComponent,
  ListItemPrimaryTextComponent,
  ListItemSecondaryTextComponent,
  ListContentPanelComponent,
  ListContentEmptyStateComponent,
  EmptyStateComponent,
  EmptyStateTitleComponent,
  EmptyStateDescriptionComponent,
  EmptyStateImageComponent,
  ListContentUnselectedStateComponent,
  ListSelectionComponent,
  ListSelectionItemComponent,
  ListSelectionSearchComponent,
  ListFilterPipe,
  ListSelectionAddButtonComponent,
  CardMainComponent,
  CardAsideComponent,
  CardTitleComponent,
  CardSectionComponent,
  CardSectionSubTitleComponent,
  CardSectionTitleComponent,
  CardActionsComponent,
  CardToolbarComponent,
  SelectContainerComponent,
  FormGroupComponent,
  SizingSmallDirective,
  SizingMediumDirective,
  SizingLargeDirective,
  SizingFullDirective,
  ErrorBannerComponent,
  InfoBannerComponent,
  ButtonGroupComponent,
  ExpandableFabComponent,
  FieldComponent,
  FieldInlineComponent,
  EditableFieldComponent,
  EditableFieldInlineComponent,
  CenterComponent,
  NestableFormDirective,
  LoadingContainerComponent,
  FooterComponent,
  ExpandableSearchComponent,
  RightAlignContainerComponent,
  ListSearchComponent,
  AccessibleClickDirective,
  AccessibleViewDirective,
  AccessibleRemoveViewDirective,
  AccessibleContextMenuViewDirective,
  ListItemRoleDirective,
  ToolTipDirective,
  AvatarComponent,
  AvatarGroupComponent,
  DisplayModeDirective,
  InputMaskComponent,
  CurrencyMaskComponent,
  CurrencyMaskDirective,
  MessagesComponent,
];

@NgModule({
  imports: [
    CommonModule,
    RouterModule,
    ReactiveFormsModule,
    MatToolbarModule,
    MatMenuModule,
    MatButtonModule,
    MatCardModule,
    MatIconModule,
    MatRippleModule,
    MatProgressSpinnerModule,
    MatNativeDateModule,
    DateRangeModule,
    MatListModule,
    ExpandableFabItemModule,
    TextListItemModule,
    DataTableContainerModule,
    MatFormFieldModule,
    MatInputModule,
    MatTooltipModule,
    TranslationModule,
    MatSidenavModule,
  ],
  declarations: components,
  exports: [
    ...components,
    DateRangeComponent,
    ExpandableFabItemComponent,
    TextListItemComponent,
    DataTableContainerComponent,
  ],
  providers: [TranslationService, CurrencyPipe],
})
export class IgniteDesignSystemModule {
  static updateTranslation(translationKey: string, translationValue: string) {
    TranslationService.updateTranslation(translationKey, translationValue);
  }

  static forRoot(): ModuleWithProviders {
    return {
      ngModule: IgniteDesignSystemModule,
      providers: [],
    };
  }

  static forChild(): ModuleWithProviders {
    return {
      ngModule: IgniteDesignSystemModule,
    };
  }
}
