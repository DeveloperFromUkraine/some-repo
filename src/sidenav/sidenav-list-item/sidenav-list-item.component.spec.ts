import { ComponentFixture, TestBed } from "@angular/core/testing";
import { SideNavListItemComponent } from './index';
import { ComponentTest } from "../../../test/test-bed/component";
import { DebugElement } from '@angular/core';
import { By } from "@angular/platform-browser";
import { EventEmitter } from "@angular/core/src/event_emitter";
import { dispatchEvent } from "@angular/core/src/view/util";

describe('Side Nav List Item', () => {
    let fixture: ComponentFixture<SideNavListItemComponent>;
    let component: SideNavListItemComponent;
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        await ComponentTest.createTestBed([], [SideNavListItemComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(SideNavListItemComponent);
        component = fixture.componentInstance;

        fixture.detectChanges();
    });

    it('should set icon', async () => {
        component.icon = 'icon';

        await fixture.detectChanges();
        ne = fixture.debugElement.query(By.css('.app-link-icon')).nativeElement;

        expect(ne.textContent).toContain(component.icon);
    });

    it('should set description', async () => {
        component.description = 'description';

        await fixture.detectChanges();
        ne = fixture.debugElement.query(By.css('.app-link-text')).nativeElement;

        expect(ne.textContent).toContain(component.description);
    });

    it('should emit onClick when click event emitted', () => {
        de = fixture.debugElement.query(By.css('.app-link'));
        ne = de.nativeElement;
        let spy = jest.fn();

        component.onClick.subscribe(spy);
        ne.dispatchEvent(new Event('click'));

        expect(spy).toHaveBeenCalled();
    });

    it('should call handleClick when keyup.enter event emitted', () => {
        jest.spyOn(component, 'handleClick');

        fixture.debugElement.triggerEventHandler('keyup.enter', { preventDefault: jest.fn() });

        expect(component.handleClick).toHaveBeenCalled();
    });

    it('should call handleClick when keyup.space event emitted', () => {
        jest.spyOn(component, 'handleClick');

        fixture.debugElement.triggerEventHandler('keyup.space', { preventDefault: jest.fn() });

        expect(component.handleClick).toHaveBeenCalled();
    });

    it('should emit onClick event with key value', () => {
        component.key = 'test';
        let spy = jest.fn();
        
        component.onClick.subscribe(spy);
        component.handleClick(new Event('keyup.enter'));

        expect(spy).toHaveBeenCalledWith(component.key);
    });

    it('should match snapshot', async () => {
        component.icon = 'mail';
        component.description = 'test description';

        await fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });
});