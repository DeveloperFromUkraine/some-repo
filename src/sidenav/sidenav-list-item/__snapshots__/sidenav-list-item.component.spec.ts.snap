// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Nav List Item should match snapshot 1`] = `
<ign-sidenav-list-item
  description={[Function String]}
  icon={[Function String]}
  onClick={[Function EventEmitter]}
  role={[Function String]}
>
  <a
    class="app-link"
  >
    
    
    <i
      aria-hidden="true"
      class="material-icons app-link-icon"
    >
      mail
    </i>
    
    
    <div
      class="app-link-text"
    >
      test description
    </div>
    
  
  </a>
</ign-sidenav-list-item>
`;
